x-pack/plugins/lens/common/expressions/rename_columns/rename_columns.ts

```diff
@@ -6,14 +6,20 @@
  */
 
 import { i18n } from '@kbn/i18n';
-import { ExpressionFunctionDefinition, Datatable, DatatableColumn } from 'src/plugins/expressions';
-import { IndexPatternColumn } from './operations';
+import {
+  ExpressionFunctionDefinition,
+  Datatable,
+  DatatableColumn,
+} from '../../../../../../src/plugins/expressions/common';
 
 interface RemapArgs {
   idMap: string;
 }
 
-export type OriginalColumn = { id: string } & IndexPatternColumn;
+type OriginalColumn = { id: string; label: string } & (
+  | { operationType: 'date_histogram'; sourceField: string }
+  | { operationType: string; sourceField: never }
+);
```

Comment:
Not entirely sure what this is about?

Reply:
Unfortunately the type `OriginalColumn` depends on all the operation definitions (it's a union of them) and moving them here it is not necessary, or useful now.
The goal of this type is to make it work correctly the `getColumnName` which has a specific logic for `date_histogram`, therefore the type has been simplified to cover that specific case.